src/python/pants/core/util_rules/external_tool.py

```diff
@@ -175,6 +226,29 @@ def get_request(self, plat: Platform) -> ExternalToolRequest:
             f"{self.known_versions}"
         )
 
+    def check_version_constraint(self, version: str) -> None:
+        if not self.constraints or self.constraints.specifier.contains(version):  # type: ignore[attr-defined]
+            # all ok
+            return None
+
+        action = self.options.use_unsupported_version
+        if action is UnsupportedVersionUsage.Allow:
+            logger.debug(
+                f"Ignoring constraints for version {version}, does not satisfy version constraints {self.constraints}"
+            )
+        elif action is UnsupportedVersionUsage.LogWarning:
+            logger.warning(
+                f"{self.name} {version} may not be compatible with this version of pants, as it does not satisfy the constraints {self.constraints}"
+            )
+        elif action is UnsupportedVersionUsage.RaiseError:
+            raise UnsupportedVersion(
+                f"{self.name} {version} does not satisfy the version constraints {self.constraints}"
+            )
+        else:
+            raise ValueError(
+                f"Internal Error: unexpected value for --use-unsupported-version: {action!r}"
+            )
```

Comment:
This is already validated by our options code.

Reply:
ah, yeah, can remove it. it actually did hit a few times while I was working on it :D (but not because the value was wrong, so was a "false" positive.. I was using `is` instead of `==` when checking the enum value.